net/tcp/tcp_send_unbuffered.c

```diff
@@ -584,7 +584,10 @@ ssize_t psock_tcp_send(FAR struct socket *psock,
       goto errout;
     }
 
-  /* If this is an un-connected socket, then return ENOTCONN */
+  /* Check early if this is an un-connected socket, if so, then
+   * return -ENOTCONN. Note, we will have to check this again, as we can't
+   * guarantee the state won't change until we have the network locked.
+   */
 
   if (psock->s_type != SOCK_STREAM || !_SS_ISCONNECTED(psock->s_flags))
```

Comment:
should we remove _SS_ISCONNECTED from the check? it is useless without net_lock.